src/main/java/use_case/preferences/PreferenceInputData.java

```diff
@@ -1,36 +1,81 @@
 package use_case.preferences;
 
 import java.util.Date;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 /**
  * The Input Data for the Signup Use Case.
  */
 public class PreferenceInputData {
-
+    private final String fullname;
     private final String username;
+    private final String password;
+    private final String image;
+    private final String repeatPassword;
+    private final String location;
+    private final String gender;
+    private final Date dateOfBirth;
+    private final List<String> preferredGender;
+    private final HashMap<String, Integer> preferredAge;
     private final List<String> tags;
     private final String bio;
     private final Map<String, Boolean> preferences;
-    private final List<String> preferredGender;
-    private final Map<String, Integer> preferredAge;
 
-    public PreferenceInputData(String username, List<String> tags, String bio,  Map<String, Boolean> preferences,
-                               List<String> preferredGender, Map<String, Integer> preferredAge) {
+    public PreferenceInputData(String fullname, String username, String password, String image, String repeatPassword,
+                               String location, String gender, Date dateOfBirth, List<String> preferredGender,
+                               HashMap<String, Integer> preferredAge, List<String> tags, String bio,  Map<String, Boolean> preferences) {
+        this.fullname = fullname;
         this.username = username;
+        this.password = password;
+        this.image = image;
+        this.repeatPassword = repeatPassword;
+        this.location = location;
+        this.gender = gender;
+        this.dateOfBirth = dateOfBirth;
+        this.preferredGender = preferredGender;
+        this.preferredAge = preferredAge;
         this.tags = tags;
         this.bio = bio;
         this.preferences = preferences;
-        this.preferredGender = preferredGender;
-        this.preferredAge = preferredAge;
     }
 
+    public String getImage() {
+        return image;
+    }
+    public String getFullname() {
+        return fullname;
+    }
 
     public String getUsername() {
         return username;
     }
 
+    public String getPassword() {
+        return password;
+    }
+
+    public String getRepeatPassword() {
+        return repeatPassword;
+    }
+
+    public String getLocation() {
+        return location;
+    }
+
+    public String getGender() {
+        return gender;
+    }
+
+    public Date getDateOfBirth() {
+        return dateOfBirth;
+    }
+
+    public List<String> getPreferredGender() { return preferredGender; }
+
```

Comment:
Style: We could also condense the getters into single-line methods to improve scrolling efficiency for collaborators, as well as make it less overwhelming & easier to review.